optics_functions/utils.py

```diff
@@ -0,0 +1,281 @@
+"""
+Utilities
+*********
+
+Reusable utilities that might be needed in multiple optics functions.
+
+"""
+import logging
+import string
+from contextlib import contextmanager
+from time import time
+from typing import Sequence, Tuple, Iterable
+
+import numpy as np
+import pandas as pd
+from tfs import TfsDataFrame
+
+from optics_functions.constants import (REAL, IMAG, PLANES, PHASE_ADV,
+                                        X, Y, S, NAME, DELTA_ORBIT)
+
+LOG = logging.getLogger(__name__)
+D = DELTA_ORBIT
+
+# DataFrames -------------------------------------------------------------------
+
+
+def prepare_twiss_dataframe(beam: int,
```

Comment:
beam seems very lhc specific and only one of the 3 options is used.. maybe call it inverted sequence or similar

Reply:
it's not easy to name, though. "inverted sequence" sounds like you should activate it for beam2... maybe `invert_signs_madx_convention` .. because it inverts not all the signs that you would expect

Reply:
better?

Reply:
yup